src/main/java/org/jabref/Logger.java

```diff
@@ -0,0 +1,153 @@
+                             final String msg, final Object... args) {
+        com.jcabi.log.Logger.error(source, msg, args);
+    }
+
+    /** end of direct pass through **/
+
+    public static void trace(final Object source, final String msg, Throwable e) {
+        com.jcabi.log.Logger.trace(source, msg);
+    }
+
+    public static void trace(
+            final Object source,
+            final String msg, final Throwable e, final Object... args
+    ) {
+        if (com.jcabi.log.Logger.isTraceEnabled(source)) {
+            com.jcabi.log.Logger.trace(source, msg + ": %[exception]s", merge(args, e));
+        }
+    }
+
+    public static void debug(final Object source, final Throwable e, final String msg) {
+        com.jcabi.log.Logger.debug(source, msg);
+    }
+
+    public static void debug(
+            final Object source,
+            final String msg, final Throwable e, final Object... args
+    ) {
+        if (com.jcabi.log.Logger.isDebugEnabled(source)) {
+            com.jcabi.log.Logger.debug(source, msg + ": %[exception]s", merge(args, e));
+        }
+    }
+
+    public static void info(final Object source, final Throwable e, final String msg) {
+        com.jcabi.log.Logger.info(source, msg);
+    }
+
+    public static void info(final Object source, final Throwable e) {
+        com.jcabi.log.Logger.info(source, "%[exception]s", e);
+    }
+
+    public static void info(
+            final Object source,
+            final String msg, final Throwable e, final Object... args
+    ) {
+        if (com.jcabi.log.Logger.isInfoEnabled(source)) {
+            com.jcabi.log.Logger.info(source, msg + ": %[exception]s", merge(args, e));
+        }
+    }
+
+    public static void warn(final Object source, final Throwable e, final String msg) {
+        com.jcabi.log.Logger.warn(source, msg);
+    }
+
+    public static void warn(final Object source, final Throwable e) {
+        com.jcabi.log.Logger.warn(source, "%[exception]s", e);
+    }
+
+    public static void warn(
+            final Object source,
+            final String msg, final Throwable e, final Object... args
+    ) {
+        if (com.jcabi.log.Logger.isWarnEnabled(source)) {
+            com.jcabi.log.Logger.warn(source, msg + ": %[exception]s", merge(args, e));
+        }
+    }
+
+    public static void error(final Object source, final Throwable e, final String msg) {
+        com.jcabi.log.Logger.error(source, msg);
+    }
+
+    public static void error(final Object source,
+                             final String msg, final Throwable e, final Object... args) {
+        // error is always logged. Therefore, no isErrorEnabled(Object) exists
+        com.jcabi.log.Logger.error(source, msg + ": %[exception]s", merge(args, e));
+    }
+
+    /**
+     * Merges args and e together into an array starting with args
+     */
+    private static Object[] merge(Object[] args, Throwable e) {
+        final int length = args.length;
```

Comment:
Guava has a method for Array concat:
https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/ObjectArrays.html#concat(T%5B%5D,%20T%5B%5D,%20java.lang.Class)